### PR TITLE
Add Tf publishing to AckermannSteering system

### LINKED ISSUE #1573: AckermannSteering System does not publish Tf
## Enviornment
OS: Ubuntu 20.04
Used with ROS2 Galactic over ign_bridge

## Details
It appears that the AckermannSteering system does not publish tf alonside its odometry output, which means that this plugin is unusable for use with slamtoolbox, as no odom->base_footprint transform is generated.

This does not occur with the DiffDrive system plugin, which does produce such an output, here in source: https://github.com/gazebosim/gz-sim/blob/d6e69d0c4ad764e6a69e952d5a129e48c19ea159/src/systems/diff_drive/DiffDrive.cc#L618

For refrence, here is where that line would be in the ackermann system, which simply publishes odom:
https://github.com/gazebosim/gz-sim/blob/d6e69d0c4ad764e6a69e952d5a129e48c19ea159/src/systems/ackermann_steering/AckermannSteering.cc#L671

It belive that simply pasting the tf code from the DiffDrive system over to the AckermannSteering system should solve this problem, but I'm not too familar with the ignition ecosystem.

### PR DESCRIPTION
**Pull request by andyblarblar**
# :tada: New feature

Closes #1573 

## Summary
Previously, the AckermannSteering system plugin did not publish tf alongside odometry, unlike the DiffDrive controller. This ment that when used with ROS, no odom->base_footprint was generated, preventing its use with SlamToolbox. This PR ports that DiffDrive tf publishing to the AckermannSteering controller, as well as the appropriate docs.

### Here's ackermann in use with slam-toolbox post fix:

Peek 2022-07-05 19-19

## Checklist
- [x] Signed all commits for DCO
- [x] Added tests
- [x] Updated documentation (as needed)
- [x] Updated migration guide (as needed)
- [x] Consider updating Python bindings (if the library has them)
- [x] `codecheck` passed (See [contributing](https://ignitionrobotics.org/docs/all/contributing#contributing-code))
- [x] All tests passed (See [test coverage](https://ignitionrobotics.org/docs/all/contributing#test-coverage))
- [x] While waiting for a review on your PR, please help review [another open pull request](https://github.com/pulls?q=is%3Aopen+is%3Apr+user%3Aignitionrobotics+repo%3Aosrf%2Fsdformat+archived%3Afalse+) to support the maintainers

**Note to maintainers**: Remember to use **Squash-Merge** and edit the commit message to match the pull request summary while retaining `Signed-off-by` messages.

### CODE REVIEW

File: src/systems/ackermann_steering/AckermannSteering.cc [hunk 1]
```diff
@@ -343,6 +346,13 @@ void AckermannSteering::Configure(const Entity &_entity,
   this->dataPtr->odomPub = this->dataPtr->node.Advertise<msgs::Odometry>(
       odomTopic);
 
+  std::string tfTopic{"/model/" + this->dataPtr->model.Name(_ecm) + 
+    "/tf"};
+  if (_sdf->HasElement("tf_topic"))
+    tfTopic = _sdf->Get<std::string>("tf_topic");
+  this->dataPtr->tfPub = this->dataPtr->node.Advertise<msgs::Pose_V>(
```
Comment (ahcorde) [inline src/systems/ackermann_steering/AckermannSteering.cc:353]:
```suggestion
  auto tfTopic = validTopic(tfTopic);
  if (tfTopic.empty())
  {
    ignerr << "AckermannSteering plugin invalid tf topic name "
           << "Failed to initialize." << std::endl;
    return;
  }

  this->dataPtr->tfPub = this->dataPtr->node.Advertise<msgs::Pose_V>(
```

File: src/systems/ackermann_steering/AckermannSteering.hh [hunk 2]
```diff
@@ -90,6 +90,22 @@ namespace systems
   /// messages. This element if optional, and the default value is
   /// `/model/{name_of_model}/odometry`.
   ///
+  /// `<tf_topic>`: Custom topic on which this system will publish the
+  /// transform from `frame_id` to `child_frame_id`. This element is optional,
+  ///  and the default value is `/model/{name_of_model}/tf`.
```
Comment (ahcorde) [inline src/systems/ackermann_steering/AckermannSteering.hh:95]:
```suggestion
  /// and the default value is `/model/{name_of_model}/tf`.
```

File: src/systems/ackermann_steering/AckermannSteering.hh [hunk 3]
```diff
@@ -90,6 +90,22 @@ namespace systems
...
+  /// origin of the odometry transform in both the `<tf_topic>`
+  /// `ignition.msgs.Pose_V` message and the `<odom_topic>`
+  /// `ignition.msgs.Odometry` message. This element if optional, and the
+  /// default value is `{name_of_model}/odom`.
+  ///
+  /// `<child_frame_id>`: Custom `child_frame_id` that this system will use as
+  /// the target of the odometry transform in both the `<tf_topic>`
+  /// `ignition.msgs.Pose_V` message and the `<odom_topic>`
+  /// `ignition.msgs.Odometry` message. This element if optional,
+  ///  and the default value is `{name_of_model}/{name_of_link}`.
```
Comment (ahcorde) [inline src/systems/ackermann_steering/AckermannSteering.hh:107]:
```suggestion
  /// and the default value is `{name_of_model}/{name_of_link}`.
```

**Review by ahcorde — CHANGES_REQUESTED** [review]

**Review by ahcorde — CHANGES_REQUESTED** [review]
Linters are failing https://github.com/gazebosim/gz-sim/runs/7208392386?check_suite_focus=true

**Comment by andyblarblar** [comment]
I've implemented the suggestions and delt with the linter and CI. The fix is still fully working so it should be all set.

**Review by ahcorde — CHANGES_REQUESTED** [review]
can you add a test to check that the tf topic is generating data ?

**Comment by andyblarblar** [comment]
Test added and passed.

File: test/integration/ackermann_steering_system.cc [hunk 4]
```diff
@@ -309,6 +310,125 @@ TEST_P(AckermannSteeringTest, IGN_UTILS_TEST_DISABLED_ON_WIN32(SkidPublishCmd))
   EXPECT_LT(poses[0].Rot().Z(), poses[3999].Rot().Z());
 }
 
+/////////////////////////////////////////////////
+TEST_P(AckermannSteeringTest, IGN_UTILS_TEST_DISABLED_ON_WIN32(TfPublishes))
+{
+  // Start server
+  ServerConfig serverConfig;
+  serverConfig.SetSdfFile(common::joinPaths(std::string(PROJECT_SOURCE_PATH),
+      "/test/worlds/ackermann_steering_slow_odom.sdf"));
```
Comment (ahcorde) [inline test/integration/ackermann_steering_system.cc:319]:
```suggestion
  serverConfig.SetSdfFile(common::joinPaths(std::string(PROJECT_SOURCE_PATH),
      "test", "worlds", "ackermann_steering_slow_odom.sdf"));
```

**Review by ahcorde — CHANGES_REQUESTED** [review]

**Review by ahcorde — DISMISSED** [review]
Uhm, focal is not happy https://github.com/gazebosim/gz-sim/runs/7385398190?check_suite_focus=true

**Comment by andyblarblar** [comment]
The test failing on CI also fails on ign-gazebo6 for me, and contains no refrence to the files changed in this PR, so it seems to be a seprerate issue.

**Review by chapulina — APPROVED** [review]
LGTM, thanks!

**Pull request merged**